sdk/data/azappconfig/client_test.go

```diff
@@ -327,3 +344,321 @@ func TestSettingWithEscaping(t *testing.T) {
+		require.NoError(t, err)
+	}
+}
+
+func TestSnapshotArchive(t *testing.T) {
+	snapshotName := "archiveSnapshotsTest" + string(testId)
+
+	client := NewClientFromConnectionString(t)
+
+	_, err := CreateSnapshot(client, snapshotName, nil)
+
+	require.NoError(t, err)
+
+	_, err = client.RecoverSnapshot(context.Background(), snapshotName, nil)
+	require.NoError(t, err)
+
+	snapshot, err := client.GetSnapshot(context.Background(), snapshotName, nil)
+
+	// Check that snapshot is in a "ready" state
+	require.NoError(t, err)
+	require.Equal(t, azappconfig.SnapshotStatusReady, *snapshot.Snapshot.Status)
+
+	// Archive the snapshot
+	update, err := client.ArchiveSnapshot(context.Background(), snapshotName, nil)
+	require.NoError(t, err)
+	require.Equal(t, azappconfig.SnapshotStatusArchived, *update.Snapshot.Status)
+
+	//Best effort snapshot cleanup
+	CleanupSnapshot(client, snapshotName)
+}
+
+func TestSnapshotRecover(t *testing.T) {
+	snapshotName := "recoverSnapshotsTest" + string(testId)
+
+	client := NewClientFromConnectionString(t)
+
+	_, err := CreateSnapshot(client, snapshotName, nil)
+	require.NoError(t, err)
+
+	snapshot, err := client.GetSnapshot(context.Background(), snapshotName, nil)
+	require.NoError(t, err)
+
+	_, err = client.ArchiveSnapshot(context.Background(), snapshotName, nil)
+	require.NoError(t, err)
+
+	// Check that snapshot is archived
+	snapshot, err = client.GetSnapshot(context.Background(), snapshotName, nil)
+	require.NoError(t, err)
+	require.Equal(t, azappconfig.SnapshotStatusArchived, *snapshot.Snapshot.Status)
+
+	// Recover the snapshot
+	update, err := client.RecoverSnapshot(context.Background(), snapshotName, nil)
+	require.NoError(t, err)
+	require.Equal(t, azappconfig.SnapshotStatusReady, *update.Snapshot.Status)
+
+	// Best effort snapshot cleanup
+	CleanupSnapshot(client, snapshotName)
+}
+
+func TestSnapshotCreate(t *testing.T) {
+	snapshotName := "createSnapshotsTest" + string(testId)
+
+	client := NewClientFromConnectionString(t)
+
+	//Create a snapshot
+	snapshot, err := CreateSnapshot(client, snapshotName, nil)
+
+	require.NoError(t, err)
+	require.Equal(t, snapshotName, *snapshot.Snapshot.Name)
+
+	// Best effort cleanup snapshot
+	CleanupSnapshot(client, snapshotName)
+}
+
+func CreateSnapshot(c *azappconfig.Client, snapshotName string, sf []azappconfig.SettingFilter) (azappconfig.GetSnapshotResponse, error) {
+	if sf == nil {
+		all := "*"
+		sf = []azappconfig.SettingFilter{
+			{
+				KeyFilter: &all,
+			},
+		}
+	}
+
+	retPer := int64(3600)
+
+	opts := &azappconfig.BeginCreateSnapshotOptions{
+		RetentionPeriod: &retPer,
+	}
+
+	//Create a snapshot
+	resp, err := c.BeginCreateSnapshot(context.Background(), snapshotName, sf, opts)
+
+	if err != nil {
+		return azappconfig.GetSnapshotResponse{}, err
+	}
+
+	if resp == nil {
+		return azappconfig.GetSnapshotResponse{}, fmt.Errorf("resp is nil")
+	}
+	_, err = resp.PollUntilDone(context.Background(), &runtime.PollUntilDoneOptions{
```

Comment:
Per offline discussion, the `Poller[T]` should return the `Snapshot` so it's returned here on terminal success. Then you don't need to `GetSnapshot()` below.